src/yoroi-wallets/cardano/transactionManager.test.ts

```diff
@@ -0,0 +1,86 @@
+import DeviceInfo from 'react-native-device-info'
+
+import {storage as rootStorage, YoroiStorage} from '../storage'
+import {Transaction} from '../types'
+import {mockApi, mockedAddressesByChunks, mockedBackendConfig, mockTx} from './mocks'
+import {makeMemosManager, makeTransactionManager} from './transactionManager'
+
+jest.mock('./api', () => mockApi)
+
+describe('transaction manager', () => {
+  DeviceInfo.getVersion = () => '9.9.9'
+
+  afterEach(rootStorage.clear)
+
+  it('stores memos', async () => {
+    const txManager = await makeTransactionManager(mockStorage, mockedBackendConfig)
+
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: mockTx})
+
+    await txManager.saveMemo(mockTx.id, 'memo 1')
+
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: {...mockTx, memo: 'memo 1'}})
+  })
+
+  it('gets updated tx cache values', async () => {
+    const txManager = await makeTransactionManager(mockStorage, mockedBackendConfig)
+
+    expect(txManager.getTransactions()).toMatchSnapshot()
```

Comment:
can you pick something from each return value to check specifically?
```
expect(transactions.length).toBe(1)
await txManager.doSync(mockedAddressesByChunks)
expect(transactions.length).toBe(2)
```

Reply:
Should the snapshots be removed?

Reply:
Or should that be added as a extra check?

Reply:
I'm fine either way, just make the tests read well